### ISSUE TITLE
"Host name must not match method 'vagrant'"

### ISSUE THREAD

**Issue by bwo** [dougm/vagrant-tramp#12]
Like so:

```
(let ((default-directory "/vagrant:box"))
  (start-file-process "foo" (current-buffer) "python" "-c" elpy-config--get-config))
```

**Comment by ryanprior** [comment]
Okay, I tracked this down and I think I understand and can explain it.

When you use the directory "/vagrant:box" it doesn't see that you're trying to use the `vagrant` method. With only one colon, the default TRAMP method is selected, which by default is ssh, but what it is specifically shouldn't matter. I guess somebody decided to disallow the default TRAMP method to connect to a host with the same name as a TRAMP method.

So it thinks you're trying to connect to `vagrant:`, which by default means `ssh:(nobody@)vagrant:` and then it's like "you cannot ssh into a host called 'vagrant' when there's a TRAMP method called 'vagrant,' put on the brakes!" and gives you the super helpful error message: _"Host name must not match method 'vagrant'"_

This to my mind is a misdirection from what the issue actually is: you triggered fall-back to the default TRAMP method by not including another colon.

As far as I can tell from my web searches and reading the TRAMP docs and source code, this is the TRAMP behavior as intended, however unhelpful that may be, and not a bug in vagrant-tramp. Therefore I'm going to close the bug for now and recommend that we open a bug in Emacs (using <kbd>M-x report-emacs-bug</kbd> or by sending an e-mail to bug-gnu-emacs@gnu.org) explaining why that behavior and error message is unhelpful. Then hopefully we can get a bunch of TRAMP users together and fix this for everybody.

If anybody knows something this package can do independently to make this situation better, then maybe there is a bug in vagrant-tramp after all, and you should feel free to open it back up for discussion.

__tl;dr to get the behavior you're looking for, you can include another colon like so:__

    (let ((default-directory "/vagrant:box:"))
    (start-file-process "foo" (current-buffer) "python" "-c" elpy-config--get-config))
